source/includes/authentication.cpp

```diff
@@ -45,4 +45,15 @@ auto client = mongocxx::client(uri);
 auto uri = mongocxx::uri("mongodb://<db_username>:<db_password>@<hostname>:<port>/?"
                          "authMechanism=PLAIN&tls=true");
 auto client = mongocxx::client(uri);
-// end-plain
\ No newline at end of file
+// end-plain
+
+// start-auth-err
+try {
+    auto uri = mongocxx::uri("mongodb://<db_username>:<db_password>@<hostname>:<port>/?"
+                         "authSource=admin&authMechanism=SCRAM-SHA-256");
+    auto client = mongocxx::client(uri);
+    client["db"].run_command(bsoncxx::from_json(R"({"ping": 1})")); // Any command requiring authorization
+} catch (const mongocxx::exception& ex) {
+    std::cerr << "Error: " << ex.what() << std::endl;
+}
```

Comment:
```suggestion
auto uri = mongocxx::uri("mongodb://<db_username>:<db_password>@<hostname>:<port>/?"
                        "authSource=admin&authMechanism=SCRAM-SHA-256");
auto client = mongocxx::client(uri);

try {
    // Any command requiring authorization
    client["db"].run_command(bsoncxx::from_json(R"({"ping": 1})"));
} catch (const mongocxx::exception& ex) {
    std::cerr << "Error: " << ex.what() << std::endl;
}
```

Recommend minimizing the scope of code within the try-catch statement to that which is expected to throw the relevant exception.